packages/esm-patient-registration-app/src/patient-registration/field/id/identifier-selection-overlay.tsx

```diff
@@ -128,81 +97,67 @@ const PatientIdentifierOverlay: React.FC<PatientIdentifierOverlayProps> = ({
             />
-            {showIdentifierSources && identifierType.checked && identifierType?.identifierSources?.length > 0 && (
-              <div className={styles.radioGroup}>
-                <RadioButtonGroup
-                  legendText={t('source', 'Source')}
-                  name={`${identifierType?.fieldName}-identifier-sources`}
-                  defaultSelected={identifier?.source?.uuid}
-                  onChange={(sourceUuid: string) => handleSelectingIdentifierSource(identifierType?.uuid, sourceUuid)}
-                  orientation="vertical">
-                  {identifierType?.identifierSources.map((source) => (
-                    <RadioButton
-                      key={source.uuid}
-                      labelText={source.name}
-                      name={source.uuid}
-                      value={source.uuid}
-                      className={styles.radioButton}
-                      disabled={
-                        isOffline &&
-                        isUniqueIdentifierTypeForOffline(identifierType) &&
-                        source.autoGenerationOption?.manualEntryEnabled
-                      }
-                    />
-                  ))}
-                </RadioButtonGroup>
-              </div>
-            )}
+            {patientIdentifier &&
+              identifierType?.identifierSources?.length > 0 &&
+              /* 
+                This check are for the cases when there's an initialValue identifier is assigned
+                to the patient
+                The corresponding flow is like:
+                1. If there's no change to the actual initial identifier, then the source remains null, 
+                hence the list of the identifier sources shouldn't be displayed.
+                2. If user wants to edit the patient identifier's value, hence there will be an initialValue,
+                along with a source assigned to itself(only if the identifierType has sources, else there's nothing to worry about), which by
+                default is the first identifierSource
+              */
+              (!patientIdentifier.initialValue || patientIdentifier?.selectedSource) && (
+                <div className={styles.radioGroup}>
+                  <RadioButtonGroup
+                    legendText={t('source', 'Source')}
+                    name={`${identifierType?.fieldName}-identifier-sources`}
+                    defaultSelected={patientIdentifier?.selectedSource?.uuid}
+                    onChange={(sourceUuid: string) => handleSelectingIdentifierSource(identifierType, sourceUuid)}
+                    orientation="vertical">
+                    {identifierType?.identifierSources.map((source) => (
+                      <RadioButton
+                        key={source.uuid}
+                        labelText={source.name}
+                        name={source.uuid}
+                        value={source.uuid}
```

Comment:
If we are controlling this component are we listening for user changes to this radio button? - mostly just regarding setting 'value' field

Reply:
Yes, the Radio Button Group handles the change.